core/embed/bootloader/main.c

```diff
@@ -347,6 +354,9 @@ int bootloader_main(void) {
     firmware_present =
         check_image_header_sig(hdr, vhdr.vsig_m, vhdr.vsig_n, vhdr.vpub);
   }
+
+  header_present = firmware_present;
```

Comment:
this is wrong; we need to keep `firmware_present` set to secfalse up until the next if-branch, otherwise in case of a glitch it could remain `sectrue` after the next check

in fact, isn't something like that true for all the previous checks too?